### ISSUE TITLE
Problems gettings started

### ISSUE THREAD

**Issue by wijerstelecom** [VincentVrijburg/moneybird-dotnet#79]
Hi Vincent,

Thank you for all your work for this Moneybird nuget package and making it public. What I have is not a general issue, but I struggle to get started with your Moneybird nuget package. Can you provide me with a simple copy & pastable and working example of c# code including authenticating to Moneybird using option 1 (API Key for personal use) of your instructions and some simple action like list sales invoices or something.

Thank you in advance.

Twan.

**Comment by VincentVrijburg** [comment]
Hi Twan,

You're welcome! Let me try to provide you with some sample code to get you started.

It seems like you have figured out already that you would like to use the simple authentication option (1), with an access token for personal use. This means you don't need to do anything fancy with OAuth, ClientIds, and callback URLs.

Something that you do need to figure out is how you get your access token into the code. This depends on the type of C# application you're trying to integrate Moneybird into, but generally speaking, the access token (which does not need to be renewed like the OAuth tokens normally do) can be stored in an `appsettings.json` or come from environment variables you pass to the application. I would highly suggest not to hard code it into your application.

In terms of getting started with calling the Moneybird API from code, you really don't need to do more than the following.
```csharp
var config = new MoneybirdConfig();
using var client = new MoneybirdClient(config); // This is the disposable version, a singleton version is also available
```

After which you can list your sales invoices.

```csharp
// ADMINISTRATION_ID = The identifier of the administration you would like to make a request for.
// ACCESS_TOKEN = The access token with the correct rights to access the administration.
var salesInvoices = await client.SalesInvoice.GetAsync("ADMINISTRATION_ID", "ACCESS_TOKEN");
```

It is that simple to list the sales invoices. Now, the `GetAsync` method on most endpoints also has an overload to send options for filtering with the request. Have a look here at the options for filtering Sales Invoices: https://developer.moneybird.com/api/sales_invoices/#get_sales_invoices

When it comes to creating and updating entities... the methods have so called `CreateOptions` and `UpdateOptions` parameters to allow the passing of an object with information to create or update an entity (e.g. SalesInvoice). In case of updating, the idea is that it only updates the information you provide with the request so you can leave out the properties you do not wish to update.

```csharp
var createOptions = new ContactCreateOptions
{
    Contact = new ContactCreateItem
    {
        CompanyName = "Test B.V.",
        // etc.
    }
};

// ADMINISTRATION_ID = The identifier of the administration you would like to make a request for.
// ACCESS_TOKEN = The access token with the correct rights to access the administration.
var newContact = await client.Contact.CreateAsync("ADMINISTRATION_ID", createOptions, "ACCESS_TOKEN");
```
I highly suggest using an IDE with proper auto-completion so you can discover all the possibilities of the filter, create, and update option objects.

When it comes to deleting entities... it is really as simple as passing the entity id, like so.

```csharp
// ADMINISTRATION_ID = The identifier of the administration you would like to make a request for.
// ENTITY_ID = The identifier of the sales invoice you would like to delete.
// ACCESS_TOKEN = The access token with the correct rights to access the administration.
var salesInvoices = await client.SalesInvoice.DeleteByIdAsync("ADMINISTRATION_ID", "ENTITY_ID", "ACCESS_TOKEN");
```

The general trend here is that you can re-use the MoneybirdConfig object and that you need your administration ID and access token with every request you make.

This package very closely aligns with the official documentation from Moneybird: https://developer.moneybird.com/
However, some features have not been implemented yet such as pagination. This means you are limited by the defaults right now as set by Moneybird: https://developer.moneybird.com/#pagination

Thank you for reminding me that I should dedicate some time to writing better documentation for this package. I have been putting the emphasis on making the package really easy to use so I could focus on implementing all the endpoints (which is not done yet), but things change rapidly so it might not always be clear how to use the package going from one version to another. Anyway, that's the downside of using a package, like this one, that is still in development.

I hope this helps! Let me know if you need more help or if you run into some issues.

**Comment by wijerstelecom** [comment]
Hi Vincent,

The value of discount=(null)

When needed i can send you the complete json response in a pm.

**Comment by VincentVrijburg** [comment]
Ok, from my understanding of the official documentation that was not possible but if you get that response it must be. Can you send the full response to me? Because there are other discount properties in the response that I would like to double-check as well.

**Comment by VincentVrijburg** [comment]
The issue has been fixed in [release 0.13.1](https://github.com/VincentVrijburg/moneybird-dotnet/releases/tag/v0.13.1)
